classic/create.go

```diff
@@ -0,0 +1,254 @@
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return "", fmt.Errorf("failed to download %s: %v", url, resp.StatusCode)
+	}
+
+	if pbar != nil {
+		pbar.Start(name, float64(resp.ContentLength))
+		mw := io.MultiWriter(w, pbar)
+		_, err = io.Copy(mw, resp.Body)
+		pbar.Finished()
+	} else {
+		_, err = io.Copy(w, resp.Body)
+	}
+
+	return w.Name(), err
+}
+
+var policyRc = []byte(`
+#!/bin/sh
+while true; do
+    case "\$1" in
+      -*) shift ;;
+      makedev) exit 0;;
+      x11-common) exit 0;;
+      *) exit 101;;
+    esac
+done
+`)
+
+var runInChroot = func(chroot string, cmd ...string) error {
+	cmdArgs := []string{"chroot", chroot}
+	cmdArgs = append(cmdArgs, cmd...)
+
+	if output, err := exec.Command(cmdArgs[0], cmdArgs[1:]...).CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to run %s in chroot %s: %s (%s)", cmd, chroot, err, output)
+	}
+
+	return nil
+}
+
+func downloadLxdRootfs(pbar progress.Meter) (string, error) {
+	url, err := findDownloadURL()
+	if err != nil {
+		return "", err
+	}
+
+	fname, err := downloadFile(url, pbar)
+	if err != nil {
+		return "", err
+	}
+
+	return fname, nil
+}
+
+func unpackLxdRootfs(fname string) error {
+	if err := os.MkdirAll(dirs.ClassicDir, 0755); err != nil {
+		return fmt.Errorf("failed to create classic mode dir: %s", err)
+	}
+
+	cmd := exec.Command("tar", "-C", dirs.ClassicDir, "-xpf", fname)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to unpack %s: %s", fname, err)
+	}
+
+	return nil
+}
+
+func customizeClassicChroot() error {
+	// create the snappy mountpoint
+	if err := os.MkdirAll(filepath.Join(dirs.ClassicDir, "snappy"), 0755); err != nil {
+		return fmt.Errorf("failed to create snappy mount point: %s", err)
+	}
+
+	// copy configs
+	for _, f := range []string{"hostname", "hosts", "timezone", "localtime"} {
+		src := filepath.Join("/etc/", f)
+		dst := filepath.Join(dirs.ClassicDir, "etc", f)
+		if err := helpers.CopyFile(src, dst, helpers.CopyFlagPreserveAll); err != nil {
+			return err
+		}
+	}
+
+	// ensure daemons do not start
+	if err := ioutil.WriteFile(filepath.Join(dirs.ClassicDir, "/usr/sbin/policy-rc.d"), policyRc, 0755); err != nil {
+		return fmt.Errorf("failed to write policy-rc.d: %s", err)
+	}
+
+	// remove ubuntu user, will come from snappy OS
+	if err := runInChroot(dirs.ClassicDir, "deluser", "ubuntu"); err != nil {
+		return err
+	}
+
+	// install extra packages; make sure chroot can resolve DNS
+	resolveDir := filepath.Join(dirs.ClassicDir, "/run/resolvconf/")
+	if err := os.MkdirAll(resolveDir, 0755); err != nil {
+		return fmt.Errorf("failed to create %s: %s", resolveDir, err)
+	}
+	src := "/run/resolvconf/resolv.conf"
+	dst := filepath.Join(dirs.ClassicDir, "/run/resolvconf/")
+	if err := helpers.CopyFile(src, dst, helpers.CopyFlagPreserveAll); err != nil {
+		return err
+	}
+
+	// enable libnss-extrausers
+	if err := runInChroot(dirs.ClassicDir, "apt-get", "install", "-y", "libnss-extrausers"); err != nil {
```

Comment:
We do, we can set DEBIAN_FRONTEND=noninteractive in the environment. But I think thats not needed because the libnss-extrausers does not use debconf. If it becomes a problem its trivial to add :)